docs/09_tutorials/10_nested_containers/index.md

```diff
@@ -0,0 +1,206 @@
+---
+content_title: Nested Containers
+link_text: Nested Containers
+---
+
+## Overview
+
+This tutorial demonstrates the usage of various C++ nested-containers in smart contracts for the following EOSIO types:
+
+* [Multi-Index Table](https://developers.eos.io/manuals/eosio.cdt/latest/how-to-guides/multi-index/index)
+* [Key-Value Map](https://developers.eos.io/manuals/eosio.cdt/latest/how-to-guides/key-value-api/kv_map/index)
+
+### The Nested-Containers Covered Are
+
+This tutorial covers the following netsted-containers combinations.
+
+* Single standard template library (STL) containers defined in the `std` namespace
+    * std::vector
+    * std::set
+    * std::optional
+    * std::map
+    * std::tuple
+    * std::pair which needs special attention for cleos input format
+* Standard C++ structures
+    * Simple struct
+    * Complex struct
+* Two-layers nested containers made of combinations of the following
+    * std::vector
+    * std::set
+    * std::optional
+    * std::map
+    * std::tuple
+    * std::pair
+* One three-layers nested container for farther extentions demonstration
+
+[[info | Note]]
+| std::array, std::variant are not supporterd yet]]
+
+### Cleos commands
+
+In adition the cleos commands for various nested-containers are also documented. Specifically it shows how to emplace and how to retrieve data using cleos *get table* and *get kv_table*. The cleos commands are documented at the end of the two shell scripts which accompany this tutorial.
+
+## Demo Source Files
+
+The source files which accompany this tutorial are located in the [eosio.cdt](https://github.com/EOSIO/eosio.cdt/tree/develop) repository, under *[tests/toolchain/build-pass](https://github.com/EOSIO/eosio.cdt/tree/develop/tests/toolchain/build-pass/)* folder and they are organized as follows: 
+
+  * The *nestcontn2a.cpp* file codes a contract `nestcontn2a` whose multi-index table `person2` has various containers
+  * The *tupletest.cpp* file codes a contract `tupletest` which shows the support for `std::tuple` for multi-index table `person2` by the new *eosio-cpp* compiler version
+  * The *nestcontn2kv.cpp* file codes a contract `nestcontn2kv` whose `eosio::kv::map` table wraps struct `person2kv` that has various containers and different types of keys
```

Comment:
nestcontn2kv.cpp ... that has various containers in the value part of eosio::kv::map